contrib/test/integration/swap.yml

```diff
@@ -0,0 +1,24 @@
+---
+
+- name: Obtain current state of swap
+  command: swapon --noheadings --show=NAME
+  register: swapon
+
+- name: Setup swap if none already, to prevent kernel firing off the OOM killer
+  block:
+
+    - name: /root/swap is truncated if not in-use
+      shell: '[ -n "$(losetup -j /root/swap)" ] || truncate -s 8G /root/swap;'
+
+    - name: A loop device is created for /root/swap if not in-use
+      shell: '[ -n "$(losetup -j /root/swap)" ] || losetup --show -f /root/swap'
+      register: losetup
+
+    - name: Swap device is formatted if losetup created a new device
+      shell: 'mkswap "$(losetup -j /root/swap | cut -d : -f 1)"'
```

Comment:
This and 22 are a little more duplication and hackery than I'm comfortable with. What would you think of changing line 14 to:

    shell: '[ -n "$(losetup -j /root/swap --noheadings --output=NAME)" ] ...

then register `losetup` always contains the loop device name, and you can reuse it safely in lines 18 and 22